simple_history/models.py

```diff
@@ -668,6 +669,19 @@ def post_delete(self, instance, using=None, **kwargs):
         else:
             self.create_historical_record(instance, "-", using=using)
 
+    def pre_delete(self, instance, **kwargs):
+        """
+        pre_delete method to ensure all deferred fileds are loaded on the model
+        """
+        # First check if instance is history enabled model
+        if not hasattr(instance._meta, "simple_history_manager_attribute"):
+            return
```

Comment:
I'm glad you slowed us down. Could we add this check too?

```python
        if not getattr(settings, "SIMPLE_HISTORY_ENABLED", True):
            return
```

Reply:
Sure!